### PR TITLE
docs: Search Metadata Batch 05

### PR DESCRIPTION
**Pull request by boruszak**
*This PR was created as part of the Consul Docs DevDot migration.*

## Consul Docs Search Refresh | Batch 05

Page metadata headers in the Consul documentation currently include “page title” and “page description” fields. Because our search functions do not match full article text, these fields are used to: 
1. Provide the terms that the internal search functions match with to produce results, meaning that these descriptions index the site’s content for search
1. Populate the internal search results, meaning that these descriptions will be displayed to users as a way to help them select the most relevant page
1. Order internal search results based on Title match and then Description match, alphabetizing within each, meaning that the words used in the page titles have a higher impact on the search results that appear first

Additionally, these fields describe pages for their external Google Search results, meaning that the first 160 characters of each description greatly impact search rankings and user decisions when selecting pages.

## Notes for Reviewers
- **Page titles** currently apply ad hoc punctuation schema in an attempt to include the most important terms and produce more meaningful top results when assuming results are returned alphabetically. Please remember that titles _will continue to change_ as we make organizational and display adjustments in DevDot. When making suggestions, I encourage you to propose different combinations of terms and punctuation as you see fit - but please be aware that making these suggestions on every single title will not be an efficient use of time. I will do my best to standardize title formatting based on any additional input you provide.
  - **Primary goals in title reviews**: 
    - Accurate categorization. For example, the title has “observability” when it should be “traffic management”
    - Highest priority terms and acronyms used
- **Page descriptions**: Descriptions were written to summarize a page’s content with an eye towards how components work in Consul, user end-goals, and what content is actually contained in the page. Additionally, every description’s first 160 characters include a complete sentence, and the character cut-off was considered semantically significant for each description (for example, if a word is cut off, it should still be comprehensible, so there aren’t lingering individual characters). _You do not have to consider character counts in your review_. Make suggestions, and I will review them in light of these constraints to fine-tune wording.
  - **Primary goals in description reviews**: 
    - Technical accuracy in descriptions
    - Includes terms that could reasonably be part of to find this page’s content
    - Does not use words that are semantically insignificant or misleading. For example, “get” is not a descriptive verb *and* it has a specific use in API requests, so it has increased potential to skew search results.

### Affected Directories
This batch contains refreshed headers for all pages in the following directories:
- /docs/k8s
- /docs/k8s/crds
- /docs/k8s/operations
- /docs/k8s/upgrades
- /docs/k8s/installation
- /docs/k8s/installation/platforms
- /docs/k8s/connect
- /docs/k8s/connect/observability

### Branch Organization
To make review more manageable, these PRs are being created and submitted in batches. Batches use branch names formatted as: `docs/search-desc-batchXX`. These branches are set to merge into a single branch named `docs/search-metadata-headers` so that all updates can be merged into the `dev-portal` branch simultaneously if desired.

Please direct any questions to @boruszak.

### CODE REVIEW

File: website/content/docs/k8s/annotations-and-labels.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 layout: docs
 page_title: Annotations and Labels
 description: >-
-  The list of available labels and annotations for running Consul on Kubernetes.
+Annotations and labels configure Consul sidecar properties and injection behavior when scheduling Kubernetes clusters. Learn about the required annotations and labels that enable Consul’s service mesh and secure upstream communication on k8s in this reference guide.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/annotations-and-labels.mdx:5]:
```suggestion
Annotations and labels configure Consul sidecar properties and injection behavior when scheduling Kubernetes clusters. Learn about the annotations and labels that enable Consul’s service mesh and secure upstream communication on k8s in this reference guide.
```
I don't believe they are required

File: website/content/docs/k8s/connect/ingress-controllers.mdx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 ---
 layout: docs
-page_title: Ingress Controller Integrations
-description: Configuring Ingress Controllers With Consul On Kubernetes
+page_title: Configure Ingress Controllers for Consul on Kubernetes
+description: >-
+Ingress controllers are pluggable components that must be configured in k8s in order to use the Ingress resource. Learn how to deploy sidecars with the controller to secure its communication with Consul, review common configuration issues, and find links to example configurations.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/ingress-controllers.mdx:5]:
```suggestion
Ingress controllers are pluggable components that must be configured in Kubernetes in order to use the Ingress resource. Learn how to deploy sidecars with the controller to secure its communication with Consul, review common configuration issues, and find links to example configurations. 
```
Did you use 'K8s' for length reasons or to target an alternate SEO term? This suggestion is just to flag this instance. Feel free to ignore.

File: website/content/docs/k8s/connect/observability/metrics.mdx [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 layout: docs
-page_title: Metrics
-description: Metrics for Consul on Kubernetes
+page_title: Configure metrics for Consul on Kubernetes
+description: >-
+Use the `connectInject.metrics` Helm values to enable Prometheus and Grafana integrations and capture metrics. Consul can collect metrics from the service mesh, sidecar proxies, agents, and gateways in a k8s cluster and then display service traffic metrics in Consul’s UI for additional observability.
 ---
 
-# Metrics
+# Configure metrics for Consul on Kubernetes
```
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/observability/metrics.mdx:8]:
```suggestion
# Configure Metrics for Consul on Kubernetes
```
I think metrics is a word we can capitalize in the title

File: website/content/docs/k8s/crds/upgrade-to-crds.mdx [hunk 4]
```diff
@@ -1,11 +1,11 @@
 ---
 layout: docs
-page_title: Upgrade An Existing Cluster to CRDs
+page_title: Upgrade Existing Clusters to Use Custom Resource Definitions
 description: >-
-  Upgrade an existing cluster to use custom resources.
+Kubernetes clusters configured with a Consul Helm chart version older than 0.30.0 require updates in order to use CRDs. Learn about the changes you need to make, as well as how to migrate a Consul config entry to a k8s CRD.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/crds/upgrade-to-crds.mdx:5]:
```suggestion
Kubernetes clusters configured with a Consul Helm chart version older than 0.30.0 require updates in order to use CRDs. Learn about upgrading to a supported Helm version and how to migrate a Consul config entry to a k8s CRD.
```

File: website/content/docs/k8s/dns.mdx [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
+page_title: Resolve Consul DNS requests in Kubernetes
 description: >-
-  One of the primary query interfaces to Consul is the DNS interface. The Consul
-  DNS interface can be exposed for all pods in Kubernetes using a stub-domain
-  configuration.
+Use a k8s ConfigMap to configure KubeDNS or CoreDNS so that you can use Consul's `<service-name>.service.consul` syntax for queries and other DNS requests. In Kubernetes, this process uses either stub-domain or proxy configuration.
 ---
 
-# Consul DNS on Kubernetes
+# Resolve Consul DNS requests in Kubernetes
```
Comment (trujillo-adam) [inline website/content/docs/k8s/dns.mdx:8]:
```suggestion
# Resolve Consul DNS Requests in Kubernetes
```
Correct?

File: website/content/docs/k8s/index.mdx [hunk 6]
```diff
@@ -1,14 +1,11 @@
...
 description: >-
-  Consul has many integrations with Kubernetes. You can deploy Consul to
-  Kubernetes using the Helm chart, sync services between Consul and Kubernetes,
-  automatically secure Pod communication with Connect, and more. This section
-  documents the official integrations between Consul and Kubernetes.
+Consul supports Kubernetes natively, allowing you to deploy Consul sidecars to a Kubernetes service mesh and sync the k8s service registry with non-k8s services. Learn how to install Consul on Kubernetes with Helm or the Consul K8s CLI and get started with tutorials.
 ---
 
-# Kubernetes
+# Consul & Kubernetes
```
Comment (trujillo-adam) [inline website/content/docs/k8s/index.mdx:8]:
```suggestion
# Consul on Kubernetes
```
Putting my foot down on the no-ampersands policy.

File: website/content/docs/k8s/index.mdx [hunk 7]
```diff
@@ -1,14 +1,11 @@
 ---
 layout: docs
-page_title: Kubernetes
+page_title: Consul & Kubernetes
```
Comment (trujillo-adam) [inline website/content/docs/k8s/index.mdx:3]:
```suggestion
page_title: Consul on Kubernetes
```

File: website/content/docs/k8s/operations/certificate-rotation.mdx [hunk 8]
```diff
@@ -1,10 +1,11 @@
 ---
 layout: docs
-page_title: Certificate Rotation
-description: Rotate Certificate on Kubernetes Cluster safely
+page_title: Rotate TLS Certificates for Consul on Kubernetes
+description: >-
+As of Consul Helm version 0.29.0, new server agent TLS certificates are issued every time the Helm version is upgraded. Learn how to trigger certificate rotation manually when it doesn't happen automatically.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/operations/certificate-rotation.mdx:5]:
```suggestion
In Consul Helm version 0.29.0 and later, new server agent TLS certificates are issued every time the Helm version is upgraded. Learn how to manually trigger certificate rotation if they do not rotate automatically.
```
I wonder if we should cut the intro statement about the version.

File: website/content/docs/k8s/operations/tls-on-existing-cluster.mdx [hunk 9]
```diff
@@ -1,10 +1,11 @@
 ---
 layout: docs
-page_title: Configure TLS on an Existing Cluster
-description: Configure TLS on an existing Consul cluster running in Kubernetes
+page_title: Rolling Updates to TLS for Existing Clusters on Kubernetes
+description: >-
+Consul's Helm chart began supporting TLS communication within clusters in version 0.16.0. Follow the instructions to trigger rolling updates for consul-k8s without causing downtime.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/operations/tls-on-existing-cluster.mdx:5]:
```suggestion
Consul Helm chart 0.16.0 and later supports TLS communication within clusters. Follow the instructions to trigger rolling updates for consul-k8s without causing downtime.
```

**Review by trujillo-adam — APPROVED** [review]
Left a few suggestions - pre-approving so you're not blocked.

File: website/content/docs/k8s/connect/connect-ca-provider.mdx [hunk 10]
```diff
@@ -1,10 +1,11 @@
----
+"---
 layout: docs
-page_title: Configuring a Connect CA Provider
-description: Configuring a Connect CA Provider
----
+page_title: Configure Certificate Authority (CA) for Consul on Kubernetes
+description: >-
+Consul includes a built-in CA, but when bootstrapping a cluster on k8s, you can configure your service mesh to use a custom certificate provider instead. Learn how to configure Vault as an external CA in primary and secondary datacenters and manually rotate Vault tokens.
+---"
```
Comment (lkysow) [inline website/content/docs/k8s/connect/connect-ca-provider.mdx:6]: trailing "

**Review by lkysow — COMMENTED** [review]
Sorry I didn't get to this in time but everything looks good, just one typo.

**Pull request merged**